### PR TITLE
add iOS library latest-legacy toggle switch

### PR DESCRIPTION
**Pull request by phantumcode**
_Issue #, if available:_

_Description of changes:_
* Create lib-legacy directory based on the current existing lib directory 
* add a v2 (latest) - v1 toggle switch for Amplify iOS V1 and V2

Screenshot:
Amplify iOS Doc Version Switch

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by tjleing — COMMENTED** [review]
it seems like `/lib/[x]` files are being replaced with `/lib-v1` files -- are redirects in place?  on iOS all of the paths should still work (but access a v2 page), but on Android links going to `/lib` pages will be broken until we add them.

File: src/components/Menu/index.tsx [hunk 1]
```diff
@@ -67,13 +67,12 @@ export default class Menu extends React.Component<MenuProps, MenuState> {
...
-    if (
-      (this.props.url.startsWith("/ui") ||
-        this.props.url.startsWith("/ui-legacy")) &&
-      this.props.filterKey !== "react-native" &&
-      this.props.filterKey !== "flutter"
+    let showLibVersionSwitcher = false
+    if ((this.props.url.startsWith("/ui") || this.props.url.startsWith("/ui-legacy")) && this.props.filterKey !== "react-native" && this.props.filterKey !== "flutter"
     ) {
       showVersionSwitcher = true;
+    } else if ((this.props.url.startsWith("/lib") || this.props.url.startsWith("/lib-legacy")) && this.props.filterKey == 'ios') {
```
Comment (tjleing) [inline src/components/Menu/index.tsx:74]:
```suggestion
    } else if ((this.props.url.startsWith("/lib") || this.props.url.startsWith("/lib-v1")) && this.props.filterKey == 'ios') {
```

[the remark this answers is not in the document]
Comment (phantumcode) [reply src/components/Menu/index.tsx]: yes, I've updated to `if` statement and added a `TODO:` comment for Android.

Comment (phantumcode) [reply src/components/Menu/index.tsx]: Thanks for catching that, updated to `lib-v1` in latest revision

[the remark this answers is not in the document]
Comment (tjleing) [reply src/components/Menu/index.tsx]: thank you!

**Comment by phantumcode** [comment]
> it seems like `/lib/[x]` files are being replaced with `/lib-v1` files -- are redirects in place? on iOS all of the paths should still work (but access a v2 page), but on Android links going to `/lib` pages will be broken until we add them.

Android docs are still available under lib and all links still works.  A clone of lib was created and moved to lib-v1 for both Android and iOS, links to lib-v1 for Android won't work until they're added.  When Android is ready to update to v2, simply modify the contents under /lib/ and enable with switcher.

**Comment by abdallahshaban557** [comment]
@phantumcode  - can you please elaborate on why this toggle is needed? 
- Are the public APIs exposed to customers significantly different between V1 and v2?
- If I want to migrate between V1 and V2, do I just have to change my project setup?

**Comment by phantumcode** [comment]
> @phantumcode - can you please elaborate on why this toggle is needed?
> 
> * Are the public APIs exposed to customers significantly different between V1 and v2?
> * If I want to migrate between V1 and V2, do I just have to change my project setup?

Amplify iOS V2 introduces major breaking changes.  All public APIs in each categories will be different syntax.  It's functionally the same as V1 but introduces Swift's new async/await structured concurrency pattern.

In order to migrate from V1 to V2, customers will need to update the API usages and update the iOS app to adopt the new Swift async/await structured concurrency.

**Comment by atierian** [comment]
Does it make sense to swap the positions of `v1` and `v2 (latest)`?
```
┌────────────────┬──────────────────┐
│                │                  │
│      v1        │    v2 (latest)   │
│                │                  │
└────────────────┴──────────────────┘
```

I see there's precedent for the way you have it [here](https://docs.amplify.aws/ui/), but it feels wrong to have reverse chronological for LTR languages here.

**Comment by phantumcode** [comment]
> Does it make sense to swap the positions of `v1` and `v2 (latest)`?
> 
> ```
> ┌────────────────┬──────────────────┐
> │                │                  │
> │      v1        │    v2 (latest)   │
> │                │                  │
> └────────────────┴──────────────────┘
> ```
> 
> I see there's precedent for the way you have it [here](https://docs.amplify.aws/ui/), but it feels wrong to have reverse chronological for LTR languages here.

Good suggestion @atierian .  @hdworld11 is also in agreement.  I've updated the latest revision to re-order the toggle switches with v2 selected by default.

**Review by tjleing — APPROVED** [review]

**Pull request merged**